src/Pages/ProductList/ProductListMain/MainContents/MainContentsItems/MainItemImgSlide/MainItemImgSlide.js

```diff
@@ -1,17 +1,124 @@
-import React, { Component } from "react";
+import React, { useState, useEffect } from "react";
 import styled from "styled-components";
+import ImgSlideMoveBtn from "./ImgSlideMoveBtn/ImgSlideMoveBtn";
+import btnSvg from "../../../../ProductListSvg/ProductListSvg";
+import CarouselSlide from "./CarouselSlide/CarouselSlide";
+import mixin from "../../../../../../Styles/mixin";
 
-export default class MainItemImgSlide extends Component {
-  render() {
-    return <ImgSlideContainer></ImgSlideContainer>;
-  }
+export default function MainItemImgSlide() {
+  const [activeImg, setActiveImg] = useState(IMG_URL.length);
+  const [activeCircle, setActiveCircle] = useState(0);
+  const [isAnimationEnd, setIsAnimationEnd] = useState(false);
+  const [animationOn, setAnimationOn] = useState(true);
+  const newImg = [...IMG_URL, ...IMG_URL, ...IMG_URL];
+
+  useEffect(() => {
+    const cardNum = IMG_URL.length;
+    const newCardNum = newImg.length;
+
+    if (!animationOn) {
+      if (activeImg === 1) setActiveImg(cardNum + 1);
+      if (activeImg === cardNum + 1) setIsAnimationEnd(false);
+      if (activeImg === newCardNum - 2) setActiveImg(cardNum * 2 - 2);
+      if (activeImg === cardNum * 2 - 2) setIsAnimationEnd(false);
+    }
+    if (activeImg === 1) {
+      setTimeout(() => setIsAnimationEnd(true), 200);
+    }
+    if (activeImg === newCardNum - 2)
+      setTimeout(() => setIsAnimationEnd(true), 200);
+  }, [activeImg, animationOn, newImg]);
+
+  useEffect(() => {
+    isAnimationEnd ? setAnimationOn(false) : setAnimationOn(true);
+  }, [isAnimationEnd]);
```

Comment:
```suggestion
  useEffect(() => {
    setAnimationOn(!isAnimationEnd);
  }, [isAnimationEnd]);
```
set 함수를 두번 쓸 필요는 없어보입니다